### PR TITLE
Solana integration tests improvement

### PR DESCRIPTION
**Pull request by onyb**
### Type of Change

- [ ] Product feature
- [ ] Bug fix
- [ ] Performance improvement
- [ ] Refactor
- [ ] Other

### Tested Environments

- [ ] Development
- [ ] Staging
- [ ] Production

### Before Requesting Review

- [ ] Does your code build cleanly without any errors or warnings?
- [ ] Have you used auto closing keywords?
- [ ] Have you added tests for new functionality?
- [ ] Have validated query efficiency for new database queries?
- [ ] Have documented new functionality in README or in comments?
- [ ] Have you squashed all intermediate commits?
- [ ] Is there a clear title that explains what the PR does?
- [ ] Have you used intuitive function, variable and other naming?
- [ ] Have you requested security and/or privacy review if needed
- [ ] Have you performed a self review of this PR?

### CODE REVIEW

File: services/payments/live_solana_test.go [hunk 1]
```diff
@@ -24,16 +24,23 @@ import (
 	must "github.com/stretchr/testify/require"
 )
 
+const (
+	splMintAddress  string = "AH86ZDiGbV1GSzqtJ6sgfUbXSXrGKKjju4Bs1Gm75AQq" // SPL mint address on devnet
+	splMintDecimals uint8  = 8                                              // SPL mint decimals on devnet
+)
```
Comment (pavelbrm) [inline services/payments/live_solana_test.go:30]: Nit: make these `const` local to the test, as they are not used anywhere else. The narrower the scope, the better. Global state must be avoided, even in tests.

File: services/payments/live_solana_test.go [hunk 2]
```diff
@@ -83,43 +90,43 @@ func TestLiveSolanaStateMachineATAMissing(t *testing.T) {
...
 	ctx = context.WithValue(ctx, ctxAuthKey{}, "some authorization from CLI")
 
-	// Should transition transaction into the Authorized state
+	// State transition: Prepared -> Authorized
 	testState.Status = paymentLib.Prepared
 	marshaledData, _ = json.Marshal(testState)
 	mockTransitionHistory.Data.UnsafePaymentState = marshaledData
 	solanaStateMachine.setTransaction(&testState)
-	newTransaction, err := Drive(ctx, &solanaStateMachine)
+	authorizedTransaction, err := Drive(ctx, &solanaStateMachine)
```
Comment (pavelbrm) [inline services/payments/live_solana_test.go:98]: Suggestion: here and everywhere else, consider using less verbose language. `authTxn` conveys the same meaning, and has a higher signal/noise ratio.

File: services/payments/live_solana_test.go [hunk 3]
```diff
@@ -83,43 +90,43 @@ func TestLiveSolanaStateMachineATAMissing(t *testing.T) {
...
+	var (
+		paidTransaction *paymentLib.AuthenticatedPaymentState
+	)
+	for {
+		paidTransaction, err = Drive(timeout, &solanaStateMachine)
+		must.Equal(t, nil, err)
+		if paidTransaction.Status == paymentLib.Paid {
+			break
+		}
+	}
```
Comment (pavelbrm) [inline services/payments/live_solana_test.go:129]: Is it guaranteed that the test reaches the desired condition? Is there a chance thing could go wrong and it gets stuck in an infinite loop? Maybe we want to put some limit here? The CI will be shared with the rest of the code base eventually, and there is no desire catching bugs in tests that could run forever.

**Review by pavelbrm — COMMENTED** [review]
Left some nitpicks.

File: services/payments/live_solana_test.go (hunk 1 above)
Comment (Sneagan) [reply services/payments/live_solana_test.go:30]: Agree in principle, but there will be/are more tests in this file that need these values so I think it makes sense to keep them.

Comment (Sneagan) [reply services/payments/live_solana_test.go:33]: On second thought, with the addition of some helper functions we are able to achieve what you suggest. :)

**Review by Sneagan — APPROVED** [review]

**Pull request merged**